### PR TITLE
use Day<Tz>

### PR DESCRIPTION
**Pull request by esheppa**
As per discussion in https://github.com/chronotope/chrono/issues/820 - I've added the `Day<Tz>` type to a branch in `chrono` and tested it out over here. If this is desired to be merged, I'll update the `chrono` dependency to point to a released version once it is released

### CODE REVIEW

**Review by dtolnay — COMMENTED** [review]
https://github.com/chronotope/chrono/pull/883 appears to have stalled. I would prefer not to keep this open indefinitely, so I'll close this PR and we can revisit if things shape up upstream. Thanks for the PR!

**Comment by esheppa** [comment]
Hi @dtolnay - I'd raised this PR as I was hoping to get some feedback before finalising #883 on `chrono` - would you prefer if that was finalised first then this was re-opened? If you are happy with the changes proposed here, I'll bring #883 up to date and finalise it with some documentation

File: README.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ use chrono::{Date, Utc};
 use std::collections::BTreeMap as Map;
 
 fn main() -> db_dump::Result<()> {
-    let mut downloads = Map::<Date<Utc>, u64>::new();
+    let mut downloads = Map::<Day<Utc>, u64>::new();
```
Comment (dtolnay) [inline README.md:63]: `Date` vs `Day` -- doesn't make a difference to me.

File: examples/industry-coefficient.rs [hunk 2]
```diff
@@ -32,7 +33,9 @@ fn main() -> db_dump::Result<()> {
         .load("./db-dump.tar.gz")?;
 
     let max_date = version_downloads.iter().map(|row| row.date).max().unwrap();
-    let start_date = max_date - Duration::weeks(6);
+    let start_date = max_date
+        .checked_sub_days(Days::new(6 * 7))
```
Comment (dtolnay) [inline examples/industry-coefficient.rs:37]: Please use the panicking `Sub` impl like before. The date parser parses a 4-digit year..... so it's impossible for this value to be 262144 BCE.

File: examples/industry-coefficient.rs [hunk 3]
```diff
@@ -32,7 +33,9 @@ fn main() -> db_dump::Result<()> {
...
+        .ok_or_else(|| err(format_args!("Unable to add six weeks to date {max_date}")))?;
```
Comment (dtolnay) [inline examples/industry-coefficient.rs:38]: This is subtracting, not adding.

File: examples/industry-coefficient.rs [hunk 4]
```diff
@@ -42,7 +45,7 @@ fn main() -> db_dump::Result<()> {
         if row.date >= start_date && row.date < max_date {
             let crate_id = versions[&row.version_id];
             let downloads = downloads.entry(crate_id).or_insert_with(Downloads::default);
-            match row.date.weekday() {
+            match row.date.date().weekday() {
```
Comment (dtolnay) [inline examples/industry-coefficient.rs:48]: I tried to check the relevant issues in chrono but I didn't find why `Day<Tz>` can't have its own `Datelike` impl like I'd expected in https://github.com/chronotope/chrono/issues/820#issuecomment-1312807339. `weekday()` has an obvious meaning on a `Day<Tz>` as do all the other methods in https://docs.rs/chrono/0.4.23/chrono/trait.Datelike.html.

**Comment by dtolnay** [comment]
Ah okay, that was not clear to me.

File: examples/industry-coefficient.rs (hunk 2 above)
Comment (esheppa) [reply examples/industry-coefficient.rs:37]: Good call - I've switched to using the `Sub` impl

File: examples/industry-coefficient.rs (hunk 3 above)
Comment (esheppa) [reply examples/industry-coefficient.rs:38]: Good catch - I've removed this as per the above comment

File: examples/industry-coefficient.rs (hunk 4 above)
Comment (esheppa) [reply examples/industry-coefficient.rs:48]: I've now implemented that over at #883 and updated this to utilize the `Datelike` impl.

**Comment by esheppa** [comment]
Thanks very much for taking the time to do a review. I've rebased and pushed the changes to https://github.com/esheppa/db-dump/tree/use-chrono-daytz however they don't seem to be showing up due to the PR being closed.

**Pull request closed without merging**